wikimedia/wikipedia-ios
# Watchlist - Add watch / unwatch flow to article and diff toolbars

**Phabricator:** 
https://phabricator.wikimedia.org/T335816

### Notes
This PR connects the watch/unwatch/update expiry actions to the article toolbar and diff toolbar.

This PR is dependent on https://github.com/wikimedia/wikipedia-ios-components/pull/22 & https://github.com/wikimedia/wikipedia-ios-data/pull/7.

### Test Steps
1. Go to an article while logged out of the app on the Staging scheme. Confirm tapping "Watch" more menu option prompts you to log in. Confirm you see a choose expiry action sheet after logging in.
2. Chose an expiry. Confirm you see a success toast indicating the new expiry time, with a button to view your watchlist.
3. Tap view watchlist. Confirm the watchlist view is pushed on.
4. Go into airplane mode and repeat some previous actions. Confirm you see a "No Internet connection" banner.
5. Go to a diff screen and repeat previous steps. Confirm watch / unwatch works as expected.

**Not-so-edge case where app is in login state but server rejects cookies**
6. Login on app, go to article. Login to same account on desktop. Then log out on desktop. Go back to app article and tap watch/unwatch. Confirm you see login prompt, and are able to complete the watch/unwatch flow after logging in.
